src/include/mempool.h

```diff
@@ -276,113 +279,57 @@ class pool_t {
 void dump(ceph::Formatter *f);
 
 
-// STL allocator for use with containers.  All actual state
-// is stored in the static pool_allocator_base_t, which saves us from
-// passing the allocator to container constructors.
-
-template<pool_index_t pool_ix, typename T>
-class pool_allocator {
-  pool_t *pool;
-  type_t *type = nullptr;
+// Compile-time decorator over STL allocators to tie them with mempool.
+template<pool_index_t pool_ix,
+	 class T,
+	 template <typename, size_t...> class Alloc = std::allocator,
+	 size_t... AllocArgs>
```

Comment:
Taking `std::integral_constants` here could make the things easier (hopefully reusing the `rebind` machinery from `std::allocator_traits`) and more elegant (less limits on decoratee's parameters).